Feature/bring back tests

mikonieminen opened the pull request:
After disabling tests before doing the first release, enable them again so that we have at least something.

mikonieminen committed:
Revert "Remove test temporarily"

This reverts commit e0010fbb2871deaace8ed037a3abcc9281124786.

mikonieminen committed:
Minor fix in README.md

Add missing eval to make new opam env effective after creating the
switch.

aronerben commented on test/test.ml:
```diff
@@ -0,0 +1,53 @@
+   * below is relative to the location of the executable under _build
+   *)
+  let json = Yojson.Basic.from_file "../../../ethereal_account.json" in
+  let username = json |> member "user" |> to_string in
```
Is this a common pattern in OCaml to have intermediate `let` bindings? Wouldn't it have been easier to directly assign these to the record that you're returning? So
```ocaml
Lwt.return {
  sender = json |> member "user" |> to_string
  ...
```

mikonieminen replied:
I'm not sure what is or is not common pattern in cases like this. Since I have that intermediary `smtp_node`, I thought this would be easier to read as it is more verbose.

joseferben replied:
I think this is idiomatic because shorthand syntax can be used to create the records easily. @aronerben Could you resolve the comment if you agree so Miko can merge this PR?

aronerben replied:
@jerben Ah yes, sorry, I thought it simply needed an approver to be able to merge it

joseferben commented on test/dune:
```diff
@@ -0,0 +1,11 @@
+(test
+ (name test)
+ (modes (byte exe))
```
Why `byte`?

mikonieminen replied:
Oops, that's an old experiment. I'll remove.

mikonieminen replied:
Removed.

mikonieminen committed:
Remove old experiment

Remove modes definition form test/dune that was left behind from an old
experiment.

mikonieminen committed:
Use field punning to shorten syntax in tests

mikonieminen committed:
Reindent test/test.ml after moving to ocamlformat
